bitrise.yml

```diff
@@ -81,53 +143,198 @@ workflows:
+
+  run_docker_integration_tests:
+    steps:
+    - git-clone@8:
+        inputs:
+        - fetch_tags: "yes"
+    - script@1:
+        title: Run integration tests and export results
         inputs:
         - content: |-
+            #!/usr/bin/env bash
+            set -exo pipefail
+
+            # Here we save the original test results and deploy directories and later on we recreate them,
+            #  because during the tests these directories are removed.
+            ORIG_BITRISE_TEST_RESULT_DIR="${BITRISE_TEST_RESULT_DIR}"
+            ORIG_BITRISE_DEPLOY_DIR="${BITRISE_DEPLOY_DIR}"
+
+            # Build and setup the current binary
+            current_bitrise="$(pwd)/_tmp/test_bitrise"
+            export CGO_ENABLED=0
+            go build -o "$current_bitrise"
+
+            export PR="" PULL_REQUEST_ID=""
+            export INTEGRATION_TEST_BINARY_PATH="$current_bitrise"
+            $current_bitrise setup
+
+            # Setup per OS test name and log file name
+            linux_only_test_name_json='{"test-name":"Linux only integration tests"}'
+            linux_only_test_log_file_name="linux_only_integration_tests"
+
+            # Run tests and generate report
+            go test -v --tags linux_only ./_tests/integration/... 2>&1 | tee "./${linux_only_test_log_file_name}.log"
+
+            mkdir -p "${ORIG_BITRISE_DEPLOY_DIR}"
+            cp "./${linux_only_test_log_file_name}.log" "${ORIG_BITRISE_DEPLOY_DIR}/${linux_only_test_log_file_name}.log"
+
+            go-junit-report -in "./${linux_only_test_log_file_name}.log" -out "${ORIG_BITRISE_DEPLOY_DIR}/${linux_only_test_log_file_name}.xml"
+
+            # Export the junit test results
+            test_results_dir="${ORIG_BITRISE_TEST_RESULT_DIR}/linux_only_integration_test_results"
+            mkdir -p "${test_results_dir}"
+            cp "${ORIG_BITRISE_DEPLOY_DIR}/${linux_only_test_log_file_name}.xml" "${test_results_dir}/${linux_only_test_log_file_name}.xml"
+            echo "${linux_only_test_name_json}" > "${test_results_dir}/test-info.json"
+
+  test_binary_build:
+    steps:
+    - git-clone@8:
+        inputs:
+        - fetch_tags: "yes"
+    - script@1:
+        title: Create snapshot binaries
+        deps:
+          brew:
+          - name: goreleaser
+        inputs:
+        - content: |
+            #!/usr/bin/env bash
             set -ex
-            exit 2
-    - script:
-        title: Skippable fail with exit code 2
-        is_always_run: true
-        is_skippable: true
+
+            goreleaser release --snapshot --rm-dist
+    - deploy-to-bitrise-io@2: { }
+
+  setup_go_junit_report:
+    steps:
+    - script@1:
+        title: Setup JUnit XML converter
         inputs:
         - content: |-
+            #!/usr/bin/env bash
             set -ex
-            exit 2
-    - script:
-        title: Skipping success
-        is_always_run: false
+            go install github.com/jstemmer/go-junit-report/v2@latest
```

Comment:
nit: would be good to pin the version

Reply:
Hmm, do we want to pin an exact version? Currently, we pin the major version 2 (`github.com/jstemmer/go-junit-report/v2`)

Reply:
Yeah, that could be enough, but I also don't see any reason not to pin an exact version. A commit hash won't disappear from GitHub (well, it could be force-pushed, but let's hope it never happens)